errorsbp/batch.go

```diff
@@ -209,3 +214,23 @@ func (e *prefixedError) Error() string {
 func (e *prefixedError) Unwrap() error {
 	return e.err
 }
+
+// BatchSize returns the size of the batch for error err.
+//
+// If err is either errorsbp.Batch or *errorsbp.Batch,
+// this function returns its Len().
+// Otherwise, it returns 1 if err is non-nil, and 0 if err is nil.
+//
+// It's useful in tests,
+// for example to verify that a function indeed returns the exact number of
+// errors that's expected to be returned.
+func BatchSize(err error) int {
+	var be Batch
+	if errors.As(err, &be) {
+		return be.Len()
+	}
+	if err == nil {
```

Comment:
🔕  shouldn't we do this first?

Reply:
`errors.As` works as expected for `nil` errors, but sure :)